libnavigation-core/src/main/java/com/mapbox/navigation/core/geodeeplink/GeoDeeplink.kt

```diff
@@ -0,0 +1,48 @@
+package com.mapbox.navigation.core.geodeeplink
+
+import com.mapbox.geojson.Point
+
+/**
+ * @param deeplink string deeplink value used to create this object
+ * @param point location coordinates specified by the deeplink, null when unknown
+ * @param placeQuery human readable description of the place
+ */
+class GeoDeeplink internal constructor(
+    val deeplink: String,
+    val point: Point?,
+    val placeQuery: String?
+) {
+
+    /**
+     * Indicates whether some other object is "equal to" this one.
+     */
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (javaClass != other?.javaClass) return false
+
+        other as GeoDeeplink
+
+        if (deeplink != other.deeplink) return false
+        if (point != other.point) return false
+        if (placeQuery != other.placeQuery) return false
+
+        return true
+    }
+
+    /**
+     * Returns a hash code value for the object.
+     */
+    override fun hashCode(): Int {
+        var result = deeplink.hashCode()
+        result = 31 * result + (point?.hashCode() ?: 0)
+        result = 31 * result + (placeQuery?.hashCode() ?: 0)
```

Comment:
```suggestion
        result = 31 * result + point.hashCode()
        result = 31 * result + placeQuery.hashCode()
```

Reply:
![Screen Shot 2022-01-12 at 10 49 56 AM](https://user-images.githubusercontent.com/3021882/149203230-dc6f76b9-6453-4253-83cc-2655e5b9730f.png)

Reply:
hmmm, maybe std lib is not a dependency of the core module 🤔 it has the extension
![image](https://user-images.githubusercontent.com/8738058/149204702-13475a4c-1fe5-4046-a879-f3e9b9946661.png)